src/rpy_diffusive_flux.cc

```diff
@@ -83,13 +105,14 @@ void RPYDiffusiveFlux::compute(std::shared_ptr<GrandPotential> grand, std::share
  
                 //RPY flux calculation   
                 const double d_ij = a_i + a_j;
-                
+                const double x_int = std::min<double>(x, max_x);
+ 
                 //To remove the concern about the lower bound value spill over the buffer sites
-                const int ig_low = std::ceil((std::abs(x-d_ij)-mesh->lower_bound())/mesh->step());
-                const int ig_high = mesh->bin(x+d_ij);
+                const double y_low = std::max(x - cutoff, (x >= 1) ? 0 : d_ij-x); 
```

Comment:
Either make these both one, or both `d_ij`. Probably `d_ij` will be easier to generalize in future!
```suggestion
                const double y_low = std::max(x - cutoff, (x >= d_ij) ? 0 : d_ij-x); 
```